src/Game/GameObjects/PlayerMobile.cs

```diff
@@ -1297,6 +1297,13 @@ private void TryOpenDoors()
                                         IsDoor(s.Graphic) && s.X == x && s.Y == y && s.Z - 15 <= z &&
                                         s.Position.Z + 15 >= z))
                     GameActions.OpenDoor();
+                if (Engine.Profile.Current.AutoOpenSecretDoors)
+                {
+                    if (World.Items.Any(s =>
```

Comment:
This is really unperformant stuff.
You parse 2x the World.Items dictionary. You MUST merge the same behaves in an unique call